src/box/sql/where.c

```diff
@@ -4634,8 +4634,10 @@ sqlWhereBegin(Parse * pParse,	/* The parser context */
 			if (op) {
 				uint32_t space_id = idx_def->space_id;
 				struct space *space = space_by_id(space_id);
-				vdbe_emit_open_cursor(pParse, iIndexCur,
-						      idx_def->iid, space);
+				if (vdbe_emit_open_cursor(pParse, iIndexCur,
```

Comment:
There are many other usages of this function which ignore the result. Would it be good to update all of them? Any why did it fail here? Is there code below which depends on having `OP_OpenSpace` and `OP_IteratorOpen` generated?

Reply:
Thank you for the review! I updated them all, however not sure if this was needed in most cases since in case of failure `is_aborted` is set to `true`. Maybe it would work even if the function would be made `void`.

The `assertion` in the issue was because there were two indexes retrieved - the PK and the secondary index. For PK the commented opcode was created, a and for secondary index no opcode was created, so assertion that checks that the created opcode does not have a comment fails.

Thank you for the review! I've updated them all, but I'm not sure if this is necessary in most cases as `is_aborted` is set to `true` in case of a failure. It might have worked even if the function had been made `void`.

The `assertion` in the issue fails was due to the fact that opcodes for two indexes were created - a PK and a secondary index. A commented opcode was generated for the PK, but no opcode was generated for the secondary index, so the assertion that the generated opcode has no comment fails because the assertion looked at opcode for PK.

Reply:
Oh, I see. Maybe indeed make the function `void` then and simply drop the assertion `p->aOp[p->nOp - 1].zComment == 0` from `vdbeVComment()`?

Reply:
Agree. Fixed!